src/saveload/vehicle_sl.cpp

```diff
@@ -841,22 +845,56 @@ class SlVehicleRoadVeh : public DefaultSaveLoadHandler<SlVehicleRoadVeh, Vehicle
 		      SLE_VAR(RoadVehicle, overtaking_ctr,       SLE_UINT8),
 		      SLE_VAR(RoadVehicle, crashed_ctr,          SLE_UINT16),
 		      SLE_VAR(RoadVehicle, reverse_ctr,          SLE_UINT8),
-		SLE_CONDDEQUE(RoadVehicle, path.td,              SLE_UINT8,                  SLV_ROADVEH_PATH_CACHE, SL_MAX_VERSION),
-		SLE_CONDDEQUE(RoadVehicle, path.tile,            SLE_UINT32,                 SLV_ROADVEH_PATH_CACHE, SL_MAX_VERSION),
+		SLEG_CONDVECTOR("path.td", rv_path_td,          SLE_UINT8,                  SLV_ROADVEH_PATH_CACHE, SL_MAX_VERSION),
+		SLEG_CONDVECTOR("path.tile", rv_path_tile,      SLE_UINT32,                 SLV_ROADVEH_PATH_CACHE, SL_MAX_VERSION),
 		  SLE_CONDVAR(RoadVehicle, gv_flags,             SLE_UINT16,                 SLV_139, SL_MAX_VERSION),
 	};
 	inline const static SaveLoadCompatTable compat_description = _vehicle_roadveh_sl_compat;
 
+	void ClearPathCache() const
+	{
+		rv_path_td.clear();
+		rv_path_tile.clear();
+	}
+
+	void SplitPathCache(RoadVehicle *rv) const
+	{
+		for (const auto &pair : rv->path) {
+			rv_path_td.push_back(pair.dir);
+			rv_path_tile.push_back(pair.tile);
+		}
+	}
+
+	void JoinPathCache(RoadVehicle *rv) const
+	{
+		size_t n = std::min(rv_path_td.size(), rv_path_tile.size());
+		if (n == 0) return;
+
+		rv->path.reserve(n);
+		for (size_t c = 0; c < n; ++c) {
+			rv->path.emplace_back(rv_path_td[c], rv_path_tile[c]);
+		}
+
+		if (IsSavegameVersionBefore(SLV_PATH_CACHE_FORMAT)) {
+			/* Path cache is now taken from back instead of front, so needs reversing. */
+			std::reverse(std::begin(rv->path), std::end(rv->path));
+		}
+	}
+
 	void Save(Vehicle *v) const override
```

Comment:
A reference probably makes more sense in this context, a nullptr vehicle doesn't make much sense. But I guess I lot of the code here is C-style passing pointers so in that sense it would be consistent. Whatever you prefer.

Same for the other functions btw.

Reply:
It's the interface to SaveLoad that uses pointers, I'm not adding to changing anything here.